src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -4024,6 +4024,20 @@ void splitAndLearnFromNonNullTest(BoundExpression operandComparedToNonNull, bool
             }
         }
 
+        protected override bool VisitInterpolatedStringComponentOfBinaryOperator(BoundInterpolatedString node, bool usesBoolReturns, bool firstPartIsUnconditional, ref LocalState shortCircuitState)
+        {
+            var result = base.VisitInterpolatedStringComponentOfBinaryOperator(node, usesBoolReturns, firstPartIsUnconditional, ref shortCircuitState);
+            SetNotNullResult(node);
+            return result;
+        }
+
+        protected override BoundBinaryOperator? PushBinaryOperatorInterpolatedStringChildren(BoundBinaryOperator node, ArrayBuilder<BoundInterpolatedString> stack)
+        {
+            var result = base.PushBinaryOperatorInterpolatedStringChildren(node, stack);
+            SetNotNullResult(node);
```

Comment:
It looks lik we only set a nullability result for the top binary op. Don't we need to set a nullability result on every child binary op as well?

Reply:
We set one for every child binary op. That's what this method is responsible for.